snippets/base/templates/base/includes/snippet_js.html

```diff
@@ -117,66 +117,61 @@
         };
     }
 
-    // gSnippetsMap polyfill, available in Firefox 22 and above.
-    var gSnippetsMap = null;
-    if (supportsLocalStorage()) {
-        // localStorage is available, so we wrap it with gSnippetsMap.
-        gSnippetsMap = {
-            set: function(key, value) {
-                localStorage[key] = value;
-            },
-            get: function(key) {
-                return localStorage[key];
-            }
-        };
-        window.gSnippetsMap = gSnippetsMap;
-    } else {
-        // localStorage isn't available, use gSnippetsMap (backed by IndexedDB).
-        gSnippetsMap = window.gSnippetsMap;
-    }
+    var gSnippetsMap = window.gSnippetsMap;
+
+    // By the time show_snippet event reaches #snippets the snippet
+    // will have finished initializing and altering the DOM. It's the
+    // time to modifyLinks() and addSnippetBlockLinks().
+    var snippets = document.getElementById('snippets');
+    snippets.addEventListener('show_snippet', function(event) {
```

Comment:
w

This event used to be created in the middle of injecting the snippet.

Now that the inject snippet block is in a function, and this only needs to happen once, and not whenever a snippet is injected, I moved it up a bit.